src/EnergyPlus/LowTempRadiantSystem.cc

```diff
@@ -2287,16 +2341,90 @@ namespace LowTempRadiantSystem {
                                              CFloRadSys(RadSysNum).CWBranchNum,
                                              CFloRadSys(RadSysNum).CWCompNum);
                 }
+                if (CFloRadSys(RadSysNum).OperatingMode != NotOperating && FirstHVACIteration) CFloRadSys(RadSysNum).updateOperatingModeHistory();
 
             } else if (SELECT_CASE_var == ElectricSystem) {
 
-            } else {
+                ElecRadSys(RadSysNum).OperatingMode = NotOperating;
+                
             }
         }
 
-        OperatingMode = NotOperating; // System is not operating or can't operate; will be reset elsewhere, if necessary
     }
 
+    void HydronicSystemBaseData::updateOperatingModeHistory()
+    {
+        // Since this is only called when the operating mode is something other than "not operating",
+        // the status from the previous system time step is what it did in the last or previous time step.
+        // So, we can update the last status of the system using this information before reseting things
+        // to "not operating".
+        this->lastOperatingMode = this->OperatingMode;
+        
+        if (DataGlobals::BeginDayFlag) {
+            // The begin day flag is set which mean this is the first time step of the day.
+            // This also means that the previous time step was the last time step of yesterday.
+            // So, the day should be the previous day, the hour should bethe last hour of the
+            // day, and the time step should be the last time step.
+            this->lastDayOfSim  = DataGlobals::DayOfSim - 1;
+            this->lastHourOfDay = int(DataGlobals::HoursInDay);
+            this->lastTimeStep  = DataGlobals::NumOfTimeStepInHour;
+        } else if (DataGlobals::BeginHourFlag) {
```

Comment:
Can you confirm the nature of these blocks?  This component will be called multiple times during the same BeginDayFlag state.  And it will be called multiple times during BeginTimeStepFlag.  We've made fixes to other component models where we had to be very careful about resetting things.  If you can just confirm that this is fully thought through, that would be great.

Reply:
Yeah, it's a bit tricky and I had to think a bit about how to do that.  What allows this to work is that this routine doesn't get called at all if !FirstHVACIteration.  Without that, it would get called multiple times.  With that condition, it should only get called the absolute first time through.  So, I think the logic is correct unless I am missing something.  Does that help?  Or am I missing something?